net/dns/resolver/tsdns_test.go

```diff
@@ -348,6 +348,9 @@ func TestResolveLocal(t *testing.T) {
 		{"ns-nxdomain", "test3.ipn.dev.", dns.TypeNS, netaddr.IP{}, dns.RCodeNameError},
 		{"onion-domain", "footest.onion.", dns.TypeA, netaddr.IP{}, dns.RCodeNameError},
 		{"magicdns", dnsSymbolicFQDN, dns.TypeA, netaddr.MustParseIP("100.100.100.100"), dns.RCodeSuccess},
+		{"via_hex", dnsname.FQDN("via-0xff.1.2.3.4."), dns.TypeAAAA, netaddr.MustParseIP("fd7a:115c:a1e0:b1a:0:ff:102:304"), dns.RCodeSuccess},
```

Comment:
oh, I assumed these would be rooted at the MagicDNS suffix, without the "." at the end here.

so the test input would be "via-0xff.1.2.3.4.corp.beta.tailscale.net." or something.

Reply:
Let's make it work both ways, rooted or not?

Reply:
I went on a bit of a rabbithole here, as I couldnt get DNS queries to come in using the search domains even though they are correctly configured. I discovered that:

 - A domain with any dot is considered fully qualified, so search domains arent consulted
 - Its technically not valid to have domains that are just numbers, so this scheme is technically invalid.

As such, it only works rootless, and everyone except us would just reject the query anyway.